projects/label-designer/src/lib/label-designer/label-settings/available-fields/available-fields.component.ts

```diff
@@ -10,23 +14,35 @@ import { FieldType, ILabelField } from '../../../label-designer.interface';
   styleUrls: ['./available-fields.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class AvailableFieldsComponent {
-
-  @Input() includePseudoFields = true;
-  @Input() placeholder = 'add';
+export class AvailableFieldsComponent implements OnChanges {
   @Input() availableFields: ILabelField[];
+  @Input() includeTextField = true;
+  @Input() placeholder = 'add';
   @Input() value = '';
 
   @Output() valueChange = new EventEmitter<ILabelField>();
 
   fieldType = FieldType;
 
+  availableFieldsWithIdx: ILabelFieldWithIdx[] = [];
+  selectedIdx: number;
+
+  ngOnChanges() {
+    this.availableFieldsWithIdx = (this.availableFields || []).map((field, idx) => ({
+      ...field,
+      idx: idx
+    }));
+    this.selectedIdx = this.availableFieldsWithIdx.filter(
+      field => field.field === this.value && !(field.type == this.fieldType.qrCode || field.type == this.fieldType.text)
+    )[0]?.idx;
```

Comment:
Use `findIndex()`:

```
this.availableFieldsWithIdx = this.availableFields?.findIndex(field =>
  field.field === this.value && !(field.type == this.fieldType.qrCode
  || field.type == this.fieldType.text)
);
```

(didn't test that but should do the trick)

Reply:
Changed it to use `findIndex`. Still couldn't get rid of the variable `this.availableFieldsWithIdx` because it's needed in the template but it looks a little better now